apiserver/bundle/bundle.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package bundle defines an API endpoint for functions dealing with bundles.
+package bundle
+
+import (
+	"strings"
+
+	"github.com/juju/bundlechanges"
+	"github.com/juju/errors"
+	"gopkg.in/juju/charm.v6-unstable"
+
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/facade"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/constraints"
+	"github.com/juju/juju/state"
+	"github.com/juju/juju/storage"
+)
+
+// init registers the Bundle facade.
+func init() {
+	common.RegisterStandardFacade("Bundle", 1, NewFacade)
+}
+
+// NewFacade creates and returns a new Bundle API facade.
+func NewFacade(_ *state.State, _ facade.Resources, auth facade.Authorizer) (Bundle, error) {
```

Comment:
(Sorry for the somewhat conflicting advice. This is a newish thing.)

Please have the RegisterStandardFacade call an *unexported* function which has this signature, and have that call NewFacade. NewFacade should not mention state.State. In other facades we are creating a "Backend" interface which contains the subset of State that that facade needs. In this case you don't even need State, so you can just not include it in the NewFacade signature.

i.e. something like:

```
common.RegisterStandardFacade("Bundle", 1, newFacade)
...
func newFacade(_ *state.State, _ facade.Resources, auth facade.Authorizer) (Bundle, error) {
    return NewFacade(auth)
}
...
func NewFacade(auth facade.Authorizer) (Bundle, error) {
...
```